File: backend/src/main/java/com/backend/ord/domain/entities/UserSession.java
package com.backend.ord.domain.entities;

import com.backend.ord.domain.entities.abstracts.EntityBase;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.Instant;

@Entity
@Table(name = "user_sessions")
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class UserSession extends EntityBase {
    @Column(name = "token", nullable = false, updatable = false, unique = true)
    private String token;

    @ManyToOne(fetch = FetchType.EAGER, optional = false, cascade = CascadeType.DETACH)
    @JoinColumn(name = "user_id", referencedColumnName = "id")
    private User user;
}

File: backend/src/main/java/com/backend/ord/domain/entities/UserSession.kt
package com.backend.ord.domain.entities

import com.backend.ord.domain.entities.abstracts.EntityBase
import jakarta.persistence.*
import org.hibernate.annotations.OnDelete
import org.hibernate.annotations.OnDeleteAction

@Entity
@Table(name = "user_sessions")
class UserSession(
    @Column(name = "token", nullable = false, updatable = false, unique = true)
    var token: String,

    @OnDelete(action = OnDeleteAction.CASCADE)
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "user_id", referencedColumnName = "id")
    var user: User
) : EntityBase()
